compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/replacements/nodes/ArrayFillNode.java

```diff
@@ -0,0 +1,202 @@
+ * questions.
+ */
+package jdk.graal.compiler.replacements.nodes;
+
+import static jdk.graal.compiler.core.common.spi.ForeignCallDescriptor.CallSideEffect.HAS_SIDE_EFFECT;
+
+import java.util.EnumSet;
+
+import org.graalvm.word.LocationIdentity;
+import org.graalvm.word.Pointer;
+
+import jdk.graal.compiler.core.common.spi.ForeignCallDescriptor;
+import jdk.graal.compiler.core.common.type.StampFactory;
+import jdk.graal.compiler.graph.NodeClass;
+import jdk.graal.compiler.lir.GenerateStub;
+import jdk.graal.compiler.nodeinfo.InputType;
+import jdk.graal.compiler.nodeinfo.NodeCycles;
+import jdk.graal.compiler.nodeinfo.NodeInfo;
+import jdk.graal.compiler.nodeinfo.NodeSize;
+import jdk.graal.compiler.nodes.FrameState;
+import jdk.graal.compiler.nodes.NamedLocationIdentity;
+import jdk.graal.compiler.nodes.StateSplit;
+import jdk.graal.compiler.nodes.ValueNode;
+import jdk.graal.compiler.nodes.memory.MemoryAccess;
+import jdk.graal.compiler.nodes.spi.NodeLIRBuilderTool;
+import jdk.vm.ci.aarch64.AArch64;
+import jdk.vm.ci.code.Architecture;
+import jdk.vm.ci.meta.JavaKind;
+
+// JaCoCo Exclude
+
+/**
+ * Fills in an array with a given constant value.
+ */
+@NodeInfo(cycles = NodeCycles.CYCLES_UNKNOWN, size = NodeSize.SIZE_UNKNOWN, allowedUsageTypes = InputType.Memory)
+public class ArrayFillNode extends MemoryKillStubIntrinsicNode
+                implements StateSplit, MemoryAccess {
+
+    public static final NodeClass<ArrayFillNode> TYPE = NodeClass.create(ArrayFillNode.class);
+    private static final ForeignCallDescriptor BOOLEAN_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Boolean);
+    private static final ForeignCallDescriptor BYTE_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Byte);
+    private static final ForeignCallDescriptor CHAR_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Char);
+    private static final ForeignCallDescriptor SHORT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Short);
+    private static final ForeignCallDescriptor INT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Int);
+    private static final ForeignCallDescriptor LONG_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Long);
+    private static final ForeignCallDescriptor FLOAT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Float);
+    private static final ForeignCallDescriptor DOUBLE_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Double);
+
+    public static final ForeignCallDescriptor[] STUBS = {
+                    BOOLEAN_ARRAY_FILL_STUB,
+                    BYTE_ARRAY_FILL_STUB,
+                    CHAR_ARRAY_FILL_STUB,
+                    SHORT_ARRAY_FILL_STUB,
+                    INT_ARRAY_FILL_STUB,
+                    LONG_ARRAY_FILL_STUB,
+                    FLOAT_ARRAY_FILL_STUB,
+                    DOUBLE_ARRAY_FILL_STUB,
+    };
+
+    protected JavaKind elementKind;
+
+    @Input protected ValueNode arrayBase;
```

Comment:
I'd suggest naming this just `array` and adding a comment like `/** Address of the array object on the Java heap. */`. The term "base" could be misleading, since we often use "base" for the address of the first array element, as opposed to the address of the whole object.